mbed-events/EventQueue.h

```diff
@@ -0,0 +1,764 @@
+     *
+     *  @param update   Function called to indicate when the queue should be
+     *                  dispatched
+     */
+    void background(mbed::Callback<void(int)> update);
+
+    /** Chain an event queue onto another event queue
+     *
+     *  After chaining a queue to a target, calling dispatch on the target
+     *  queue will also dispatch events from this queue. The queues use
+     *  their own buffers and events must be handled independently.
+     *
+     *  A null queue as the target will unchain the existing queue.
+     *
+     *  The chain function allows multiple event queuest to be composed,
+     *  sharing the context of a dispatch loop while still being managed
+     *  independently
+     *
+     *  @param target   Queue that will dispatch this queue's events as a
+     *                  part of its dispatch loop
+     */
+    void chain(EventQueue *target);
+
+    /** Post an event to the queue
+     *
+     *  The specified callback will be executed in the context of the event
+     *  queue's dispatch loop.
+     *
+     *  The call function is irq safe and can act as a mechanism for moving
+     *  events out of irq contexts.
+     *
+     *  @param f        Function to execute in the context of the dispatch loop
+     *  @param a0..a4   Arguments to pass to the callback
+     *  @return         A unique id that represents the posted event and can
+     *                  be passed to cancel, or an id of 0 if there is not
+     *                  enough memory to allocate the event.
+     */
+    template <typename F>
+    int call(F f) {
+        struct local {
+            static void call(void *p) { (*static_cast<F*>(p))(); }
+            static void dtor(void *p) { static_cast<F*>(p)->~F(); }
+        };
+
+        void *p = equeue_alloc(&_equeue, sizeof(F));
+        if (!p) {
+            return 0;
+        }
+
+        F *e = new (p) F(f);
+        equeue_event_dtor(e, &local::dtor);
+        return equeue_post(&_equeue, &local::call, e);
+    }
+
+    template <typename F, typename A0>
+    int call(F f, A0 a0) {
+        return call(context10<F, A0>(f, a0));
+    }
+
+    template <typename F, typename A0, typename A1>
+    int call(F f, A0 a0, A1 a1) {
+        return call(context20<F, A0, A1>(f, a0, a1));
+    }
+
+    template <typename F, typename A0, typename A1, typename A2>
+    int call(F f, A0 a0, A1 a1, A2 a2) {
+        return call(context30<F, A0, A1, A2>(f, a0, a1, a2));
+    }
+
+    template <typename F, typename A0, typename A1, typename A2, typename A3>
+    int call(F f, A0 a0, A1 a1, A2 a2, A3 a3) {
+        return call(context40<F, A0, A1, A2, A3>(f, a0, a1, a2, a3));
+    }
+
+    template <typename F, typename A0, typename A1, typename A2, typename A3, typename A4>
+    int call(F f, A0 a0, A1 a1, A2 a2, A3 a3, A4 a4) {
+        return call(context50<F, A0, A1, A2, A3, A4>(f, a0, a1, a2, a3, a4));
+    }
+
+    /** Post an event to the queue after a specified delay
```

Comment:
Same inconsistency as before: `post` vs `call`. 
The documentation is also incorrect, the event will be post immediately but execution will happens after x ms.